boards/arm/96b_carbon_nrf51/96b_carbon_nrf51-pinctrl.dtsi

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+&pinctrl {
+	i2c0_default: i2c0_default {
+		group1 {
+			psels = <NRF_PSEL(TWIM_SDA, 0, 28)>,
+				<NRF_PSEL(TWIM_SCL, 0, 2)>;
+		};
+	};
+
+	i2c0_sleep: i2c0_sleep {
+		group1 {
+			psels = <NRF_PSEL(TWIM_SDA, 0, 28)>,
+				<NRF_PSEL(TWIM_SCL, 0, 2)>;
+			low-power-enable;
+		};
+	};
+
+	uart0_default: uart0_default {
+		group1 {
+			psels = <NRF_PSEL(UART_TX, 0, 29)>,
+				<NRF_PSEL(UART_RX, 0, 11)>;
+		};
+	};
+
+	uart0_sleep: uart0_sleep {
+		group1 {
+			psels = <NRF_PSEL(UART_TX, 0, 29)>,
+				<NRF_PSEL(UART_RX, 0, 11)>;
+			low-power-enable;
+		};
+	};
+
+	spi1_default: spi1_default {
+		group1 {
+			psels = <NRF_PSEL(SPIS_SCK, 0, 7)>,
+				<NRF_PSEL(SPIS_MOSI, 0, 0)>,
+				<NRF_PSEL(SPIS_MISO, 0, 30)>,
+				<NRF_PSEL(SPIS_CSN, 0, 25)>;
+		};
+	};
+
+	spi1_sleep: spi1_sleep {
+		group1 {
+			psels = <NRF_PSEL(SPIS_SCK, 0, 7)>,
```

Comment:
The spi_nrfx_spis driver does not currently support PM_DEVICE APIs, so I think we should not provide sleep states for "nordic,nrf-spis" nodes to avoid confusion.

Reply:
thanks, removed

Reply:
@anangl now build results in failure:

```
west build -b 96b_carbon_nrf51 tests/benchmarks/footprints -p -- -DCONFIG_PM_DEVICE=y
...
eFiles/drivers__spi.dir/spi_nrfx_spis.c.obj -c /home/gmarull/ws/nordic/zephyrproject/zephyr/drivers/spi/spi_nrfx_spis.c
In file included from /home/gmarull/ws/nordic/zephyrproject/zephyr/include/toolchain.h:50,
                 from /home/gmarull/ws/nordic/zephyrproject/zephyr/include/init.h:10,
                 from /home/gmarull/ws/nordic/zephyrproject/zephyr/include/device.h:29,
                 from /home/gmarull/ws/nordic/zephyrproject/zephyr/include/drivers/spi.h:24,
                 from /home/gmarull/ws/nordic/zephyrproject/zephyr/drivers/spi/spi_nrfx_spis.c:7:
/home/gmarull/ws/nordic/zephyrproject/zephyr/include/toolchain/gcc.h:77:36: error: static assertion failed: "/soc/spi@40004000 defined without required pin configuration"
   77 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert(EXPR, "" MSG)
      |                                    ^~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/zephyr/soc/arm/nordic_nrf/common/./soc_nrf_common.h:207:2: note: in expansion of macro 'BUILD_ASSERT'
  207 |  BUILD_ASSERT((IS_ENABLED(CONFIG_PINCTRL) &&   \
      |  ^~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/zephyr/drivers/spi/spi_nrfx_spis.c:272:2: note: in expansion of macro 'NRF_DT_ENSURE_PINS_ASSIGNED'
  272 |  NRF_DT_ENSURE_PINS_ASSIGNED(SPIS(idx), sck_pin);         \
      |  ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/gmarull/ws/nordic/zephyrproject/zephyr/drivers/spi/spi_nrfx_spis.c:319:1: note: in expansion of macro 'SPI_NRFX_SPIS_DEVICE'
  319 | SPI_NRFX_SPIS_DEVICE(1);
      | ^~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /home/gmarull/.local/bin/cmake --build /home/gmarull/ws/nordic/zephyrproject/zephyr/build
```

Reply:
Yes, I need to further extend the NRF_DT_ENSURE_PINS_ASSIGNED macro so that a driver can indicate that it does not support PM_DEVICE and thus it does not need `pinctrl-1`.

Reply:
I've updated #43925. You can cherry-pick the changes here to make CI pass.

Reply:
rebased since your PR got merged